snuba/web/views.py

```diff
@@ -92,23 +96,42 @@ def check_clickhouse() -> bool:
     Checks if all the tables in all the enabled datasets exist in ClickHouse
     """
     try:
-        for name in get_enabled_dataset_names():
-            dataset = get_dataset(name)
-            for entity in dataset.get_all_entities():
-                for storage in entity.get_all_storages():
-                    clickhouse = storage.get_cluster().get_query_connection(
-                        ClickhouseClientSettings.QUERY
-                    )
-                    clickhouse_tables = clickhouse.execute("show tables")
-                    source = storage.get_schema()
-                    if isinstance(source, TableSchema):
-                        table_name = source.get_table_name()
-                        if (table_name,) not in clickhouse_tables:
-                            return False
+        datasets = [get_dataset(name) for name in get_enabled_dataset_names()]
+        entities = itertools.chain(
+            *[dataset.get_all_entities() for dataset in datasets]
+        )
+        storages = list(
+            itertools.chain(*[entity.get_all_storages() for entity in entities])
+        )
+
+        cluster_grouped_table_names: Dict[ConnectionId, List[str]] = defaultdict(list)
```

Comment:
This can be a `MutableMapping[ConnectionId, Set[str]]`
- MutableMapping is more abstract than Dict, it provides a more restrictive interface that does not surface too many implementation details.
- Set instead of list because if the same table was here twice (which could in theory happen) you don't care about dealing with it twice.